### PR TITLE
Rotate image panel by 90°

### LINKED ISSUE #484: Rotate 90° without cropping
See https://forum.image.sc/t/couple-of-suggestions-about-omero-figure/72675

"In the preview tab, for rectangle images, rotating the image crops a part of it because of the fix canvas. Therefore, it is not possible to flip the image by 90° and having it entirely displayed. I think it could be great to have an option to flip the image by 90° and 180° (left and right)."

This feature exists on various photo apps, often with a single button that rotates the image by 90° each time it's clicked, with icons like:

Screenshot 2022-10-13 at 09 46 37
Screenshot 2022-10-13 at 09 48 20

OMERO.figure could do this by automatically setting the rotation value and updating the width/height at the same time.

### PR DESCRIPTION
**Pull request by Rdornier**
Fixes #484 

The rotation button at the bottom rotates the image/canvas by steps of 90°

image

### CODE REVIEW

**Comment by will-moore** [comment]
The display looks a bit strange for me:

Screenshot 2024-10-03 at 12 32 18

Maybe try to remove `height: 200px` from the `.rotation-controls-shown` styles and remove `absolute` from the new button.

There's a space between the 90 and `°` which makes the button bigger than it needs to be. Also I wonder if a different icon would distinguish it from the rotation icon above? (see below). Or possibly just a `+ 90°` is better?

E.g.

```
--- a/src/css/figure.css
+++ b/src/css/figure.css
@@ -650,11 +650,9 @@
     }
 
     .rotation-controls-shown .panel-rotation {
-        margin-top: 170px;
+        margin-top: 160px;
         padding: 2px;
-        min-width: 50px;
         display: block;
-        position: absolute;
         pointer-events: all;
     }
 
@@ -884,7 +882,6 @@
         background-color: #FAFAFA;
         border: 1px solid #CCCCCC;
         border-radius: 3px;
-        height: 200px;
         padding: 2px;
         position: absolute;
         z-index: 100;
diff --git a/src/templates/image_display_options.template.html b/src/templates/image_display_options.template.html
index 420c4f04..5369a15b 100644
--- a/src/templates/image_display_options.template.html
+++ b/src/templates/image_display_options.template.html
@@ -8,8 +8,8 @@
 
         <input type="range" class="rotation-slider" max="360" value="<%= rotation %>"></input>
         <button type="button" class="btn btn-outline-secondary btn-sm panel-rotation" title="Panel rotation">
-            <i class="bi-arrow-clockwise" style="font-size: 14px"></i>
-            <span>90</span> &deg;
+            <i class="bi-arrow-90deg-right" style="font-size: 14px"></i>
+            <span>90</span>&deg;
         </button>
```

**Comment by will-moore** [comment]
There's some strange behaviour when the panel is rectangular.
I didn't notice that you're also updating with width and height until I tried a rectangular panel...
But when I try to Undo the change, the width is restored but not the height.

E.g. if I take a panel like on the left and I rotate it 90° it looks like the panel in the centre. Then if I `Undo` that change, it looks like the panel on the right. The Width is undone, but not the height.

Screenshot 2024-10-03 at 13 04 41

File: src/js/models/panel_model.js [hunk 1]
```diff
@@ -299,6 +299,23 @@
...
+            var panelRotationAngle =  (parseInt(rotation) + 90) % 360
+
+            var viewport = this.getViewportAsRect()
+            var width = this.get('width')
+            var height = this.get('height')
+            var offset = (viewport.height - viewport.width)/2
+            var offset_x = viewport.x-offset
+            var offset_y = viewport.y+offset
+
+            this.cropToRoi({'x': offset_x, 'y': offset_y, 'width': viewport.height, 'height': viewport.width, 'rotation': panelRotationAngle})
```
Comment (will-moore) [inline src/js/models/panel_model.js:313]: It seems to me that you don't need this `cropToRoi` function here?, and removing it fixes the Undo issue.

Comment (Rdornier) [reply src/js/models/panel_model.js:313]: I found it necessary for rectangular big images. If we don't crop and simply update width/height, then the image appears more zoomed-in, but for non-big images, then removing this method works properly (see below). As I already had some weird issues, regarding big images, which you couldn't reproduce, can you test on a rectangular big image if you have the same issue ?

image

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#201](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/201/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/201/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - .gitignore
  - PR #584 Rdornier 'Link url to image'
    - .gitignore

--conflicts

**Comment by will-moore** [comment]
It looks like `cropToRoi()` was just needed for getting the new zoom, so I copied the lines for that and used them directly.

This is working for me with rectangular images etc.

```
+++ b/src/js/models/panel_model.js
@@ -303,15 +303,14 @@
             var rotation = this.get('rotation') || "0"
             var panelRotationAngle =  (parseInt(rotation) + 90) % 360
 
+            // For rectangular images we need to recalculate the zoom;
             var viewport = this.getViewportAsRect()
             var width = this.get('width')
             var height = this.get('height')
-            var offset = (viewport.height - viewport.width)/2
-            var offset_x = viewport.x-offset
-            var offset_y = viewport.y+offset
-
-            this.cropToRoi({'x': offset_x, 'y': offset_y, 'width': viewport.height, 'height': viewport.width, 'rotation': panelRotationAngle})
-            this.save({'rotation': panelRotationAngle, 'height': width, 'width': height});
+            var xPercent = this.get('orig_width') / viewport.height;
+            var yPercent = this.get('orig_height') / viewport.width;
+            var zoom = Math.min(xPercent, yPercent) * 100;
+            this.save({'rotation': panelRotationAngle, 'height': width, 'width': height, 'zoom': zoom});
 
             return panelRotationAngle
```

**Comment by Rdornier** [comment]
You're right, it is the zoom level that causes the issue. I've corrected it. Thanks !

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#221](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/221/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/221/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#222](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/222/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/222/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#223](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/223/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/223/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#224](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/224/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/224/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#225](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/225/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/225/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#226](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/226/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/226/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#227](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/227/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/227/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#228](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/228/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/228/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#229](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/229/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/229/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#230](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/230/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/230/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#231](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/231/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/231/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#233](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/233/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/233/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#234](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/234/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/234/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#235](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/235/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/235/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#236](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/236/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/236/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#237](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/237/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/237/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#238](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/238/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/238/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#239](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/239/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/239/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#240](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/240/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/240/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#241](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/241/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/241/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#242](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/242/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/242/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#243](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/243/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/243/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#244](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/244/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/244/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#252](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/252/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/252/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#253](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/253/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/253/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#254](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/254/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/254/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#255](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/255/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/255/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#256](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/256/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/256/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/templates/image_display_options.template.html
    - src/js/views/right_panel_view.js

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#257](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/257/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/257/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#258](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/258/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/258/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-plugins-push#259](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/259/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/259/consoleText) for more details.
Possible conflicts:
  - PR #578 Rdornier 'Image outline'
    - src/js/views/right_panel_view.js
  - PR #591 MinaEnayat 'Add Horizontal and Vertical Image Flipping Functionality'
    - src/js/views/right_panel_view.js
    - src/templates/image_display_options.template.html

--conflicts

**Comment by will-moore** [comment]
Both those conflicting PRs are merged now @Rdornier

**Comment by Rdornier** [comment]
Thanks @will-moore for merging the PRs.
I've solved the conflicts now

**Review by will-moore — APPROVED** [review]
Working fine on merge-ci now, Thanks

**Pull request merged**